SConstruct

```diff
@@ -179,18 +177,43 @@ for k in platform_opts.keys():
     for o in opt_list:
         opts.Add(o)
 
-for x in module_list:
-    module_enabled = True
-    tmppath = "./modules/" + x
-    sys.path.insert(0, tmppath)
+# Detect modules.
+opts.Add("custom_modules", "Directory path containing custom modules to build along with built-in modules.", "")
```

Comment:
Move this further up with other options (under "Advanced options" or "Components" I guess, as you prefer).

Reply:
Yeah this was my initial course of action, it's just that I've figured if we add module-specific options dynamically like `module_x_enabled`, it kinda made sense to move this option down there, but I think it's better to move this up for visibility then.